acceptance-tests.sh

```diff
@@ -1,144 +1,222 @@
-binding_guid_allowpublic=$(cf service-key "${SERVICE_INSTANCE_NAME}" "${SERVICE_KEY_NAME_ALLOWPUBLIC}" --guid)
-
-if [ "${binding_guid}" != "${client_id}" ]; then
-  echo "Incorrect client id ${client_id}; expected ${binding_guid}"
-fi
-
-if [ "${binding_guid_b}" != "${client_id_b}" ]; then
-  echo "Incorrect client id ${client_id_b}; expected ${binding_guid_b}"
-fi
-
-if [ "${binding_guid_allowpublic}" != "${client_id_allowpublic}" ]; then
-  echo "Incorrect client id ${client_id_allowpublic}; expected ${binding_guid_allowpublic}"
-fi
-
-# Client exists in UAA
-uaac client get "${binding_guid}"
-uaac client get "${binding_guid_b}"
-
-# Check that allowpublic is set on client
-uaac client get "${binding_guid_allowpublic}" | grep "allowpublic"
-
-# Delete service instance
-cf delete-service-key -f "${SERVICE_INSTANCE_NAME}" "${SERVICE_KEY_NAME}"
-cf delete-service-key -f "${SERVICE_INSTANCE_NAME}" "${SERVICE_KEY_NAME_B}"
-cf delete-service-key -f "${SERVICE_INSTANCE_NAME}" "${SERVICE_KEY_NAME_ALLOWPUBLIC}"
-cf delete-service -f "${SERVICE_INSTANCE_NAME}"
-
-# Client does not exist in UAA
-if uaac client get "${binding_guid}"; then
-  echo "Unexpectedly found client ${binding_guid} in UAA"
-  exit 1
-fi
-
-if uaac client get "${binding_guid_b}"; then
-  echo "Unexpectedly found client ${binding_guid_b} in UAA"
-  exit 1
-fi
-
-if uaac client get "${binding_guid_allowpublic}"; then
-  echo "Unexpectedly found client ${binding_guid_allowpublic} in UAA"
-  exit 1
-fi
-
-# Ensure service instance is deleted
-teardown() {
-  cf delete-service-key -f "${SERVICE_INSTANCE_NAME}" "${SERVICE_KEY_NAME}" || true
-  cf delete-service-key -f "${SERVICE_INSTANCE_NAME}" "${SERVICE_KEY_NAME_B}" || true
-  cf delete-service-key -f "${SERVICE_INSTANCE_NAME}" "${SERVICE_KEY_NAME_ALLOWPUBLIC}" || true
-  cf delete-service -f "${SERVICE_INSTANCE_NAME}" || true
+cf login -a $CF_API_URL -u $CF_USERNAME -p $CF_PASSWORD -o $CF_ORGANIZATION -s $CF_SPACE &> /dev/null
+
+uaac target $UAA_API_URL &> /dev/null
+uaac token client get $UAA_CLIENT_ID -s $UAA_CLIENT_SECRET &> /dev/null
+
+function test_cloud-gov-service-account-plan () {
+  local plan=$1 # from config.json
+  local role=$2 # in cloud foundry: space_developer | space_auditor
+
+  local passed_count=0
+  local fail_count=0
+  local results_file="cloud-gov-service-account-plan_${plan}"
+  rm -f $results_file
+
+  local svc_instance_name="${plan}-acceptance-instance"
+  local svc_key_name="${plan}-acceptance-key"
+
+  # Create service 
+  cf create-service "cloud-gov-service-account" "$plan" "$svc_instance_name" &> /dev/null
+  cf create-service-key "$svc_instance_name" "$svc_key_name" &> /dev/null
+
+  # Service Key GUID should be the username
+  local svc_key_guid=$(cf service-key "$svc_instance_name" "$svc_key_name" --guid)
+  local username=$(cf curl "/v3/service_credential_bindings/${svc_key_guid}/details" | jq -r '.credentials.username')
```

Comment:
What ensures that the username is the service key GUID? The broker just defaults the username to the GUID of the service?